File: src/org/webfabric/sitemesh/OriginalServletPath.java
package org.webfabric.sitemesh;

import javax.servlet.http.HttpServletRequest;

class OriginalServletPath implements Path {
    private final String value;

    public OriginalServletPath(HttpServletRequest request) {
        this.value = (String) request.getAttribute("javax.servlet.include.servlet_path");
    }

    public OriginalServletPath(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: src/org/webfabric/servlet/OriginalServletPath.scala
package org.webfabric.servlet;

import javax.servlet.http.HttpServletRequest
import org.webfabric.io.HierarchicalPath;

class OriginalServletPath(v: String) extends HierarchicalPath(v)

object OriginalServletPath {
  def apply(request: HttpServletRequest):OriginalServletPath = {
    return new OriginalServletPath(request.getAttribute("javax.servlet.include.servlet_path").asInstanceOf[String]);
  }
}
